### PR TITLE
Track Changes updates

### PR DESCRIPTION
**Pull request by whutchinson98**
**NOTE** [libreofficekit](https://github.com/coparse-inc/libreofficekit/pull/3) PR needs to be merged in first

### CODE REVIEW

File: src/electron/office/office_client.cc [hunk 1]
```diff
@@ -176,10 +177,13 @@ std::string OfficeClient::GetLastError() {
...
 v8::Local<v8::Value> OfficeClient::LoadDocument(v8::Isolate* isolate,
                                                 const std::string& path) {
+  office_->setOptionalFeatures(
+      LibreOfficeKitOptionalFeatures::LOK_FEATURE_NO_TILED_ANNOTATIONS);
+
   lok::Document* doc = GetDocument(path);
 
   if (!doc) {
-    doc = office_->documentLoad(path.c_str(), "en-US");
+    doc = office_->documentLoad(path.c_str(), "Language=en-US");
```
Comment (chase) [inline src/electron/office/office_client.cc:186]: Nice catch, but try to keep unrelated changes to a minimum in case we have to revert things.

It looks like they changed it to support a bunch of options. I'm not sure if they're actually included until the upstream changes are pulled though:
https://github.com/CollaboraOnline/online/blob/749fed14e0099141e72ae4b250686dbd67335616/kit/Kit.cpp#L1377-L1417

File: README.md [hunk 2]
```diff
@@ -58,3 +60,7 @@ scripts/e vs_devenv
 ## How do I pull the upstream changes from Electron?
 
 Run `scripts/pull-upstream-changes.sh`
+
+## How do I update LibreOfficeKit after compiling it locally?
+
+Run `scripts/local-lok.sh`
```
Comment (chase) [inline README.md:66]: The name of the script is a little vague out of the context of the README. Maybe `update-local-lok.sh`?

File: scripts/local-lok.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+echo "Removing existing libreofficekit"
+
+rm -rf ./src/out/Default/libreofficekit
+mkdir ./src/out/Default/libreofficekit
+
+echo "Bringing over local libreofficekit"
+
+cp -r ./../libreofficekit/libreoffice-core/instdir/* ./src/out/Default/libreofficekit/
```
[the remark this answers is not in the document]
Comment (chase) [reply scripts/local-lok.sh:10]: This also makes quite a few assumptions:

That the current directory is the top level directory.
That the build directory is Default.
That the project has been built at least once.

At the very least, these assumptions should be noted in a comment in the script.

Alternatively, they could be addressed by simply extending the `e` script which handles a lot of the setup for you.
See `scripts/e gen_clangd` code for an example:
https://github.com/coparse-inc/electron-libreoffice/blob/main/scripts/e#L253-L259

**Review by chase — CHANGES_REQUESTED** [review]

File: src/electron/office/office_client.cc (hunk 1 above)
Comment (whutchinson98) [reply src/electron/office/office_client.cc:186]: I'll create a separate PR for this

File: scripts/local-lok.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (whutchinson98) [reply scripts/local-lok.sh:10]: > Should probably be `-rf` so that it overwrites old files.

Since I remove the directory before hand `cp -r` should always be correct. I'll do `-rf` just to be safe though

[the remark this answers is not in the document]
Comment (whutchinson98) [reply scripts/local-lok.sh]: Updated the script to be more robust. But the scripts/e SCRIPT_DIR and BUILD_DIR are actually only working if you execute them from the root of the repo. SCRIPT_DIR is always set to `pwd` since the `BASH_SOURCE[0]` fails for me at least.

File: scripts/e [hunk 4]
```diff
@@ -280,4 +282,25 @@ case "$1" in
...
+  update_local_lok)
+    OUT_DIR="$BASE_PATH/src/out/$ELECTRON_OUT_DIR"
+
+    if [ -z "$LOK_PATH" ]
+    then
+      echo "LOK_PATH not set. Defaulting to $DEFAULT_LOK_PATH"
+      LOK_PATH=$DEFAULT_LOK_PATH
+    else
+      echo "LOK_PATH was set to $LOK_PATH"
+    fi
```
Comment (chase) [inline scripts/e:294]: Can do this with one line:

`LOK_PATH="${LOK_PATH:-$DEFAULT_LOK_PATH}"`

In fact you can get rid of DEFAULT_LOK_PATH and just make it `LOK_PATH="${LOK_PATH:-../libreofficekit}"`

**Review by chase — CHANGES_REQUESTED** [review]
